packages/atlas/src/components/_auth/SignUpModal/SignUpModal.tsx

```diff
@@ -248,28 +228,28 @@ export const SignUpModal = () => {
         () => {
           if (syncState.current === 'tried') {
             syncState.current = 'synced'
-            handlePasswordStepSubmit(signUpFormData.password)
+            handlePasswordStepSubmit(signUpFormData.current.password)
           }
           syncState.current = 'synced'
         }
       )
+
+      if (newMemberId) {
+        memberRef.current = newMemberId
+      }
+
       // in case of block sync logic failure assume member is synced after 10s
       setTimeout(() => {
         if (syncState.current === 'tried') {
           syncState.current = 'synced'
-          handlePasswordStepSubmit(signUpFormData.password)
+          handlePasswordStepSubmit(signUpFormData.current.password)
         }
         syncState.current = 'synced'
-      }, 20_000)
-
-      if (newMemberId) {
-        setMemberId(newMemberId)
-      }
+      }, 15_000)
```

Comment:
... and 15s here. My approach would have beef to introduce the const (so we can easily adjust it later), but if not, then let's just make them the same